### PR TITLE
[202012][sonic-utilities] CLI support for port auto negotiation

### PR DESCRIPTION
**Pull request by vdahiya12**
1. Add CLI support for port auto negotiation feature.
2. Add db_migrator change for auto negotiation feature
2. Add unit test cases for all changes

1. Add new subcommands to "config interface" command group to allow user configuring port auto negotiation
2. Add new subcommands to "show interfaces" command group to allow user show auto negotiation status
3. In db_migrator.py, change auto negotiation related DB field to latest one

Adding this PR because the cherry-pick from original PR #1568 
has conflicts 

#### How to verify it

Ran the changes on Arista7050cx3 testbed 

```
admin@sonic:~$ sudo config interface autoneg Ethernet96 enabled
admin@sonic:~$ echo $?
0
```

```
  Interface    Auto-Neg Mode    Speed    Oper    Admin
-----------  ---------------  -------  ------  -------
  Ethernet0              N/A     100G    down     down
  Ethernet4              N/A     100G    down       up
  Ethernet8              N/A     100G    down       up
 Ethernet12              N/A     100G    down       up
 Ethernet16              N/A     100G    down       up
 Ethernet20              N/A     100G    down     down
 Ethernet24         disabled     100G    down       up
 Ethernet28              N/A     100G    down       up
 Ethernet32              N/A     100G    down       up
 Ethernet36              N/A     100G    down       up
 Ethernet40              N/A     100G    down     down
 Ethernet44              N/A     100G    down       up
 Ethernet48              N/A     100G    down       up
 Ethernet52              N/A     100G    down       up
 Ethernet56              N/A     100G    down       up
 Ethernet60              N/A     100G    down       up
 Ethernet64              N/A     100G    down     down
 Ethernet68              N/A     100G    down     down
 Ethernet72              N/A     100G    down       up
 Ethernet76              N/A     100G    down       up
 Ethernet80              N/A     100G    down       up
 Ethernet84              N/A     100G    down     down
 Ethernet88              N/A     100G    down       up
 Ethernet92              N/A     100G    down       up
 Ethernet96          enabled     100G      up       up
Ethernet100              N/A     100G    down       up
Ethernet104              N/A     100G    down       up
Ethernet108              N/A     100G    down       up
Ethernet112              N/A     100G    down     down
Ethernet116              N/A     100G    down       up
Ethernet120              N/A     100G    down       up
Ethernet124              N/A     100G    down     down
Ethernet128              N/A      10G    down     down
Ethernet132              N/A      10G    down     down
```

```
admin@sonic:~$ sudo config interface autoneg Ethernet96 disabled
admin@sonic:~$ echo $?
0
```

```
admin@sonic:~$ show interfaces autoneg status
  Interface    Auto-Neg Mode    Speed    Oper    Admin
-----------  ---------------  -------  ------  -------
  Ethernet0              N/A     100G    down     down
  Ethernet4              N/A     100G    down       up
  Ethernet8              N/A     100G    down       up
 Ethernet12              N/A     100G    down       up
 Ethernet16              N/A     100G    down       up
 Ethernet20              N/A     100G    down     down
 Ethernet24          enabled     100G    down       up
 Ethernet28              N/A     100G    down       up
 Ethernet32              N/A     100G    down       up
 Ethernet36              N/A     100G    down       up
 Ethernet40              N/A     100G    down     down
 Ethernet44              N/A     100G    down       up
 Ethernet48              N/A     100G    down       up
 Ethernet52              N/A     100G    down       up
 Ethernet56              N/A     100G    down       up
 Ethernet60              N/A     100G    down       up
 Ethernet64              N/A     100G    down     down
 Ethernet68              N/A     100G    down     down
 Ethernet72              N/A     100G    down       up
 Ethernet76              N/A     100G    down       up
 Ethernet80              N/A     100G    down       up
 Ethernet84              N/A     100G    down     down
 Ethernet88              N/A     100G    down       up
 Ethernet92              N/A     100G    down       up
 Ethernet96         disabled     100G    down       up
Ethernet100              N/A     100G    down       up
Ethernet104              N/A     100G    down       up
Ethernet108              N/A     100G    down       up
Ethernet112              N/A     100G    down     down
Ethernet116              N/A     100G    down       up
Ethernet120              N/A     100G    down       up
Ethernet124              N/A     100G    down     down
Ethernet128              N/A      10G    down     down
Ethernet132              N/A      10G    down     down
```

### CODE REVIEW

**Review by prsunny — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@prsunny @lguohan please don't merge yet.
As the feature is based on Nvidia contribution on 202106 we need to be sure the relevant SAI supports it before merging the capabilities and run all the available tests before that.
Also, can you please confirm the full feature is backported and not just the CLI? Can you please share list of backported PRs which required for this feature?

File: config/main.py
Comment (lguohan) [inline config/main.py:2542]: keep on autoneg command, we do not need other commands.

File: config/main.py
Comment (lguohan) [inline config/main.py:2577]: please remove, it will not work for 202012.

File: config/main.py
Comment (lguohan) [inline config/main.py:2607]: please remove

File: config/main.py
Comment (vdahiya12) [reply config/main.py]: fixed

File: config/main.py
Comment (vdahiya12) [reply config/main.py]: fixed

File: config/main.py
Comment (vdahiya12) [reply config/main.py]: fixed

**Comment by lguohan** [comment]
are we able to resolve the alert?

**Review by lguohan — APPROVED** [review]

**Comment by vdahiya12** [comment]
> are we able to resolve the lgtm alert?

This LGTM, is for mock purposes and needs to be fix. 
Because this is in portconfig, not in a test file. We need to eliminate this if os.environ["UTILITIES_UNIT_TESTING"] == "1": check from the source files, and move all of this logic into the test files. 
I will work on it separately. For now it should be OK.

**Review by prsunny — COMMENTED** [review]
Sorry that I couldn't post the comment earlier.

**Pull request merged**
